hack/targetgen/main.go

```diff
@@ -0,0 +1,183 @@
+	if err != nil {
+		log.Fatal("failed creating the adapter templates")
+		log.Fatal(err)
+		return
+	}
+
+	// read newtarget_lifecycle.go
+	path = *cfgDir + "/pkg/apis/targets/v1alpha1/" + temp.Kind + "_lifecycle.go"
+	err = temp.replaceTemplates("scaffolding/pkg/apis/targets/v1alpha1/newtarget_lifecycle.go", path)
+	if err != nil {
+		log.Fatal("failed creating the newtarget_lifecycle.go template")
+		log.Fatal(err)
+		return
+	}
+
+	// read newtarget_types.go
+	path = *cfgDir + "/pkg/apis/targets/v1alpha1/" + temp.Kind + "_types.go"
+	err = temp.replaceTemplates("scaffolding/pkg/apis/targets/v1alpha1/newtarget_types.go", path)
+	if err != nil {
+		log.Fatal("failed creating the newtarget_types.go template")
+		log.Fatal(err)
+		return
+	}
+
+	// populate reconciler directory
+	// read adapter.go
+	path = *cfgDir + "/pkg/targets/reconciler/" + temp.Kind + "/adapter.go"
+	err = temp.replaceTemplates("scaffolding/pkg/targets/reconciler/newtarget/adapter.go", path)
+	if err != nil {
+		log.Fatal("failed creating the reconciler templates")
+		log.Fatal(err)
+		return
+	}
+
+	// read controller_test.go
+	path = *cfgDir + "/pkg/targets/reconciler/" + temp.Kind + "/controller_test.go"
+	err = temp.replaceTemplates("scaffolding/pkg/targets/reconciler/newtarget/controller_test.go", path)
+	if err != nil {
+		log.Fatal("failed creating the controller_test.go template")
+		log.Fatal(err)
+		return
+	}
+
+	// read controller.go
+	path = *cfgDir + "/pkg/targets/reconciler/" + temp.Kind + "/controller.go"
+	err = temp.replaceTemplates("scaffolding/pkg/targets/reconciler/newtarget/controller.go", path)
+	if err != nil {
+		log.Fatal("failed creating the controller.go template")
+		log.Fatal(err)
+		return
+	}
+
+	// read reconciler.go
+	path = *cfgDir + "/pkg/targets/reconciler/" + temp.Kind + "/reconciler.go"
+	err = temp.replaceTemplates("scaffolding/pkg/targets/reconciler/newtarget/reconciler.go", path)
+	if err != nil {
+		log.Fatal("failed creating the reconciler.go template")
+		log.Fatal(err)
+		return
+	}
+
+	// populate the config directory
+	// read 301-newtarget.yaml.go
+	path = *cfgDir + "/config/301-" + temp.Kind + ".yaml"
+	temp.replaceTemplates("scaffolding/config/301-newtarget.yaml", path)
+
+	fmt.Println("done")
+	fmt.Println("Next Steps:")
+	fmt.Println("Update `cmd/triggermesh-controller/main.go`")
+	fmt.Println("Update `config/500-controller.yaml`")
+	fmt.Println("Update `pkg/api/targets/v1alpha1/register.go`")
+	fmt.Printf("Create kodata symlinks in cmd/%s", temp.Kind)
+	fmt.Println("")
+	fmt.Println("Run `make codegen`")
+}
+
+type component struct {
+	Kind      string
+	TitleCase string
+	FullCaps  string
+}
+
+func (a *component) replaceTemplates(filename, outputname string) error {
+	tmp, err := template.ParseFiles(filename)
+	if err != nil {
+		return err
+	}
+
+	file, err := os.Create(outputname)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	err = tmp.Execute(file, a)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Since this is the last function call, and it returns what you need, you can return the result directly:

```suggestion
	return tmp.Execute(file, a)
```